app/controllers/patients_controller.rb

```diff
@@ -550,6 +549,120 @@ def update_monitoring_fields(patient, params, household, propagation)
+          History.follow_up_flag_edit(history_data)
+        else
+          # Handle case where follow-up flag update failed
+          error_message = 'Unable to update Follow-up Flag.'
+          render(json: { error: error_message }, status: :bad_request) && return
+        end
+      end
+    end
+
+    # Grab the patient IDs of houshold members to also update
+    apply_to_household_ids = find_household_ids(patient, params)
+    # If not applying to household, return
+    return if apply_to_household_ids.empty?
+
+    # Update selected group members if applying to household and ids are supplied
+    if clear_flag
+      apply_to_household_ids.each do |id|
+        member = current_user.get_patient(id)
+        next if member.nil?
+
+        clear_flag_reason = params.permit(:clear_flag_reason)[:clear_flag_reason]
+        clear_follow_up_flag(member, clear_flag_reason)
+      end
+    else
+      apply_to_household_ids.each do |id|
+        member = current_user.get_patient(id)
+        next if member.nil?
+
+        history_data[:patient] = member
+        history_data[:follow_up_reason_before] = member.follow_up_reason
+        history_data[:follow_up_note_before] = member.follow_up_note
+
+        # Handle success or failure of updating a follow-up flag
+        ActiveRecord::Base.transaction do
+          # Apply and save updates to the db
+          if member.update!(follow_up_reason: follow_up_reason, follow_up_note: follow_up_note)
+            # Create history item on successful update
+            History.follow_up_flag_edit(history_data)
+          else
+            # Handle case where follow-up flag update failed
+            error_message = 'Unable to update Follow-up Flag for a household member.'
+            render(json: { error: error_message }, status: :bad_request) && break
+          end
+        end
+      end
+    end
+  end
+
+  # Clear the patient's follow-up flag reason and/or note and create corresponding History items.
+  #
+  # patient - The Patient to update.
+  # clear_flag_reason - The note to include in the history item
+  def clear_follow_up_flag(patient, clear_flag_reason)
+    # Prep data needed to create history items based on this update
+    history_data = {
+      created_by: current_user.email,
+      patient: patient,
+      clear_flag_reason: clear_flag_reason,
+      follow_up_reason_before: patient.follow_up_reason
+    }
+
+    # Handle success or failure of clearing a follow-up flag
+    ActiveRecord::Base.transaction do
+      # Apply and save updates to the db
+      if patient.update!(follow_up_reason: nil, follow_up_note: nil)
+        # Create history item on successful update
+        History.clear_follow_up_flag(history_data)
+      else
+        # Handle case where follow-up flag clear failed
+        error_message = 'Unable to clear Follow-up Flag.'
+        render(json: { error: error_message }, status: :bad_request) && return
+      end
```

Comment:
The `else` case will never be hit, because `update!` will throw an error if it fails. We do want an error to be thrown so the transaction rolls back. I think it would be best to just remove the `else` case.